packages/playwright-core/src/client/waiter.ts

```diff
@@ -35,7 +35,7 @@ export class Waiter {
   constructor(channelOwner: ChannelOwner<channels.EventTargetChannel>, event: string) {
     this._waitId = createGuid();
     this._channelOwner = channelOwner;
-    this._savedZone = zones.currentZone();
+    this._savedZone = zones.current().without('apiZone');
```

Comment:
When discussed this [before](https://github.com/microsoft/playwright/pull/33409#discussion_r1830328363), we decided to try and keep all the filtering logic inside `zones.ts`. I'd stick with that and avoid passing 'apiZone' in every call site.

Reply:
This is the whole point. The API layer decides that the callback it will call **should not** have an api zone. It cannot decide so for any other type of zone data. Therefore, it's own decision is to remove apiZone data from here, and no other layer should care about that.

Reply:
The idea was to contain this logic to a single place, but I'm not too attached too it.